jupyter/restore_raft_quorum_with_tablets.ipynb

```diff
@@ -0,0 +1,680 @@
+   "execution_count": null,
+   "id": "c8156024",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for node in base_alive_nodes:\n",
+    "    node.run_cqlsh(\"DELETE value FROM system.scylla_local WHERE key = 'raft_group0_id'\")\n",
+    "    node.run_cqlsh(\"TRUNCATE system.discovery\")\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2ed22247",
+   "metadata": {},
+   "source": [
+    "## Step 8.\n",
+    "Set the new scylla.yaml parameter, recovery_leader, to Host ID of the recovery leader on each live node"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "95cc628e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "recovery_host_id = recovery_coordinator.host_id\n",
+    "for node in base_alive_nodes:\n",
+    "    node.remoter.sudo(f\"\"\"echo 'recovery_leader: {recovery_host_id}' | sudo tee -a /etc/scylla/scylla.yaml\"\"\")\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08f23e58",
+   "metadata": {},
+   "source": [
+    "## Step 9.\n",
+    "Rolling restart all live nodes, but the recovery leader must be restarted first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "545d32ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "cluster.restart_scylla(nodes=[recovery_coordinator])\n",
+    "cluster.restart_scylla(nodes=[node for node in base_alive_nodes if node != recovery_coordinator])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4691849",
+   "metadata": {},
+   "source": [
+    "## Step 10.\n",
+    "Remove/replace all dead nodes\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "617dc05e",
+   "metadata": {},
+   "source": [
+    "### step 10.1\n",
+    "Remove stopped node in alive dc. it is possible because each rack had 2 nodes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22807189",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "verification_node.run_nodetool(f\"removenode {stopped_node_host_id} --ignore-dead-nodes {','.join(dead_node_host_ids)}\")\n",
+    "cluster.nodes.remove(host_id_node_map[stopped_node_host_id])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88f56cae",
+   "metadata": {},
+   "source": [
+    "### step 10.2\n",
+    "Replace all nodes in dead dc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adc25935",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# print(list(dead_node_host_ids_maps.keys())) \n",
+    "# k = list(dead_node_host_ids_maps.keys())[0]\n",
+    "# del dead_node_host_ids_maps[k]\n",
+    "# print(list(dead_node_host_ids_maps.keys())) \n",
+    "\n",
+    "replace_nodes_by_host_id(cluster, dead_node_host_ids_maps, verification_node)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5b04413",
+   "metadata": {},
+   "source": [
+    "#### Step 10.3\n",
+    "Add new nodes to racks\n"
```

Comment:
I don't understand why you added this step. In step 10.2 we already replace all dead nodes. Why do we need more nodes?

Doing this as a test (for example to verify that topology operations work) is fine, but then you should specify that it is an additional test and not a part of the procedure.